rest/route/agent_test.go

```diff
@@ -25,6 +26,69 @@ var (
 	taskSecret = "tasksecret"
 )
 
+func TestAgentFetchExpansionsForTask(t *testing.T) {
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, rh *fetchExpansionsForTaskHandler){
+		"FactorySucceeds": func(ctx context.Context, t *testing.T, rh *fetchExpansionsForTaskHandler) {
+			copied := rh.Factory()
+			assert.NotZero(t, copied)
+			_, ok := copied.(*fetchExpansionsForTaskHandler)
+			assert.True(t, ok)
+		},
+		"ParseSucceeds": func(ctx context.Context, t *testing.T, rh *fetchExpansionsForTaskHandler) {
+			req, err := http.NewRequest(http.MethodGet, "https://example.com/rest/v2/agent/task/t1/fetch_vars", nil)
+			require.NoError(t, err)
+			req = gimlet.SetURLVars(req, map[string]string{"task_id": "t1"})
+			assert.NoError(t, rh.Parse(ctx, req))
+			assert.Equal(t, rh.taskID, "t1")
+		},
+		"RunSucceeds": func(ctx context.Context, t *testing.T, rh *fetchExpansionsForTaskHandler) {
+			rh.taskID = "t2"
+			resp := rh.Run(ctx)
+			require.NotZero(t, resp)
+			assert.Equal(t, http.StatusOK, resp.Status())
+			data, ok := resp.Data().(apimodels.ExpansionVars)
+			require.True(t, ok)
+			assert.Equal(t, data.PrivateVars, map[string]bool{"b": true})
+			assert.Equal(t, data.Vars, map[string]string{"a": "1", "b": "3"})
+		},
+	} {
+		t.Run(tName, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			require.NoError(t, db.ClearCollections(task.Collection, model.ProjectRefCollection, model.ProjectVarsCollection, model.VersionCollection))
+			t2 := task.Task{
+				Id:      "t2",
+				Project: "p1",
+				Version: "aaaaaaaaaaff001122334456",
+			}
+			pRef := model.ProjectRef{
+				Id: "p1",
+			}
+			vars := &model.ProjectVars{
+				Id:          "p1",
+				Vars:        map[string]string{"a": "1", "b": "3"},
+				PrivateVars: map[string]bool{"b": true},
+			}
+			v2 := &model.Version{
+				Id:         "aaaaaaaaaaff001122334456",
```

Comment:
Shoudl this test be for (or include) the alias parameter case?

Reply:
After responding to the first CR the parameters now get cached on the `Version` at the time of finalizing, so there isn't any new change `FetchExpansionsForTask` as part of this PR. I did keep this test from the initial implementation since we might as well up our code coverage where we can.  I've added another case here where the `Version` has params and checks if they are picked up correctly.

Reply:
Ah yeah that's what I was looking for, thanks!